napari/components/layerlist.py

```diff
@@ -62,6 +62,26 @@ def selected(self):
         """List of selected layers."""
         return [layer for layer in self if layer.selected]
 
+    @property
+    def active(self):
+        """Active layer, if any, given by top most selected."""
+        active_layer = None
+        # if multiple layers are selected sets the active layer to None
+        for layer in self:
+            if active_layer is None and layer.selected:
+                active_layer = layer
+            elif active_layer is not None and layer.selected:
+                active_layer = None
+                break
+        return active_layer
```

Comment:
As I'm looking at problems with performance my enthusiasm diminishes for hiding seme "expensive calculation" in a property. 
Maybe there should be a policy that property should not contain linear or some more expensive calculations. Then people will see. Ok, this is a function, so may not be cheap in usage. 
I think about asymptotic complexity cost. 

I strongly suggest adding some cache structure in `LayerList`. Some set, or dictionary.

Reply:
But there are requests for support O(1000) layers. Which causes call of this at least ten times slower.

Reply:
We're incredibly far from O(1000) layers right now, many many things would likely need to change, so we'd likely need to rewrite a lot of this regardless.

Reply:
The vispy scene graph is not currently designed to handle so many independent visuals, and changing that will be a big big lift, so I see O(1000) as not currently realistic

Reply:
I add print to `active` property. Then I add 10 layers. I count 60 calls of active property. I do a few checks. It looks that currently, each adds layer operation cause 6 times calls of `active`. The same happens on the change of the active layer by click on the interface.  

Maybe print increase time.
```
%time viewer.layers.active
Aaaaa 846
CPU times: user 196 µs, sys: 106 µs, total: 302 µs
Wall time: 226 µs
Out[9]: <Image layer 'el [123]' at 0x7f83a7a782e0>

len(viewer.layers)
Out[10]: 131
```